airflow/providers/amazon/aws/hooks/ec2.py

```diff
@@ -58,7 +127,7 @@ def get_instance_state(self, instance_id: str) -> str:
         :return: current state of the instance
         :rtype: str
         """
-        return self.get_instance(instance_id=instance_id).state["Name"]
+        return self.get_instances(instance_ids=[instance_id])[0][self.STATE][self.NAME]
```

Comment:
If the instanced ID doesn't exist this probably returns a very unhelpful error (`list index out of range` for example.)

Please fix this, and and a unit test case covering this.

Reply:
`get_instances` would fail in that case with following error:
```
botocore.exceptions.ClientError: An error occurred (InvalidInstanceID.Malformed) when calling the DescribeInstances operation: Invalid id: "<instance_id>"
```